drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -35,22 +35,13 @@
 #include "debug.h"
 
 
-static void _irq_handler(void *arg)
+void at86rf2xx_setup(at86rf2xx_t *dev, spi_t spi, spi_speed_t spi_speed,
+                     gpio_t cs_pin, gpio_t int_pin, gpio_t sleep_pin,
+                     gpio_t reset_pin)
 {
-    msg_t msg;
-    at86rf2xx_t *dev = (at86rf2xx_t *) arg;
-
-    /* tell driver thread about the interrupt */
-    msg.type = GNRC_NETDEV_MSG_TYPE_EVENT;
-    msg_send(&msg, dev->mac_pid);
-}
-
-int at86rf2xx_init(at86rf2xx_t *dev, spi_t spi, spi_speed_t spi_speed,
-                   gpio_t cs_pin, gpio_t int_pin,
-                   gpio_t sleep_pin, gpio_t reset_pin)
-{
-    dev->driver = &at86rf2xx_driver;
+    netdev2_t *netdev = (netdev2_t *)dev;
 
+    netdev->driver = &at86rf2xx_driver;
```

Comment:
The other `netdev2`s are also the setting the driver in the setup and it makes more sense to me. Why expose the driver when it other stuff needs to be set for the device too.

It was your idea to use this kind of inheritance ;P